client/src/procedures/primitives.rs

```diff
@@ -1024,3 +1035,33 @@ impl AesKeyWrapDecrypt {
         Ok(plaintext)
     }
 }
+
+/// This procedure is to be used to check for values inside the vault.
+/// By it's very nature, this procedure is not secure to use and is by default
+/// inactive. it MUST NOT be used in production setups.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[cfg(feature = "insecure")]
+pub struct CheckingProcedure {
```

Comment:
I find `CheckingProcedure` to be too generic of a name, it's not clear what it does. I think I still prefer `CompareSecret`.